### ISSUE TITLE
How to include a custom model?

### ISSUE THREAD

**Issue by sheldonhull** [AbdBarho/stable-diffusion-webui-docker#60]
Was working through the scripts, and found that there are explicit mountings in `mount.sh` for the apps. 

I tried adding a custom entry but couldn't get the model to show up in the UI. 
Is there something beyond adding a new entry into `mount.sh` that is required?

Example: 

```shell
#!/bin/bash

set -e

declare -A MODELS

ROOT=/stable-diffusion/src

MODELS["${ROOT}/gfpgan/experiments/pretrained_models/GFPGANv1.3.pth"]=GFPGANv1.3.pth
MODELS["${ROOT}/realesrgan/experiments/pretrained_models/RealESRGAN_x4plus.pth"]=RealESRGAN_x4plus.pth
MODELS["${ROOT}/realesrgan/experiments/pretrained_models/RealESRGAN_x4plus_anime_6B.pth"]=RealESRGAN_x4plus_anime_6B.pth
MODELS["/latent-diffusion/experiments/pretrained_models/model.ckpt"]=LDSR.ckpt
MODELS["${ROOT}/waifu-diffusion/experiments/pretrained_models/waifu-diffusion.ckpt"]=waifu-diffusion.ckpt # 👈👈👈👈👈👈👈👈👈👈👈👈👈👈

MODELS_DIR=/cache/models

```

**Comment by sheldonhull** [comment]
I'm working through it with adding to downloads, and removing my download taskfile command since you've built this into the compose process. Lots of interesting different approaches here that I think are useful to give me new ideas! 

I'll see if this works and update for others.

**Comment by AbdBarho** [comment]
@sheldonhull `hlky` UI? I don't think you can add custom models to it, I think that works for ESRGAN in `auto` only.
Can you link to documentation / description of the feature you are trying to use?

**Comment by AbdBarho** [comment]
@sheldonhull Ah, I see, waifu-difussion is just a fine tuned-model.

In theory, you don't need to change any code, you can just
```bash
mv cache/models/model.ckpt cache/models/model.backup.ckpt 
mv waifu-diffusion.ckpt cache/models/model.ckpt
```
and run the UI, any UI should work because this is just the weights.

**Comment by sheldonhull** [comment]
So this replaces the model.ckpt that's there, meaning it's not a switch for me, it's one or the other? 

I plan on learning more on the technical side of this later, so sorry if this is obvious stuff. I'm only dabbling in this area to learn a bit and have some fun ;-)

**Comment by AbdBarho** [comment]
@sheldonhull correct, it is one or the other. if you want a switch through an env variable, update the [CMD of `services/hlky/Dockerfile`](https://github.com/AbdBarho/stable-diffusion-webui-docker/blob/c1fa2f145731633916a2ab3ade7bf8cad2d7ca71/services/hlky/Dockerfile#L60) and delete the `--ckpt /cache/models/model.ckpt` 

and then you can define the model you want [in the cli args of the ui](https://github.com/AbdBarho/stable-diffusion-webui-docker/blob/c1fa2f145731633916a2ab3ade7bf8cad2d7ca71/docker-compose.yml#L31)
```bash
- CLI_ARGS=--optimized-turbo --ckpt /cache/models/waifu-diffusion.ckpt
```
You are going to have to restart the container for that to work.


If you want to update the model in runtime while the container is running, then you should probably create a feature request in the respective ui repo.

**Comment by AbdBarho** [comment]
Fixed in #75
